serpentTools/tests/test_microxs.py

```diff
@@ -0,0 +1,182 @@
+        parent = 922350
+        energy = 2.53000E-08
+        fp = 531350
+        excpectedIndYield = numpy.array([2.92737E-02])
+        excpectedCumYield = numpy.array([6.28187E-02])
+        actualIndYield, actualCumYield = self.reader.getFY(parent, energy, fp)
+        numpy.testing.assert_equal(excpectedIndYield, actualIndYield)
+        numpy.testing.assert_equal(excpectedCumYield, actualCumYield)
+
+    def test_getFY(self):
+        """Test the getFY method."""
+        # raises exception due to non-existing parent
+        with self.assertRaises(SerpentToolsException):
+            actualIndYield, actualCumYield = \
+                self.reader.getFY(500000, 2.53000E-08, 531350)
+
+        # closest energy check
+        actualIndYield, actualCumYield = self.reader.getFY(922350, 1E-05, 531350)
+        excpectedIndYield = numpy.array([2.92737E-02])
+        excpectedCumYield = numpy.array([6.28187E-02])
+        numpy.testing.assert_equal(excpectedIndYield, actualIndYield)
+        numpy.testing.assert_equal(excpectedCumYield, actualCumYield)
+
+        # non-existing daughter
+        actualIndYield, actualCumYield = self.reader.getFY(922350, 1E-05, 9999)
+        excpectedIndYield = numpy.array([])
+        excpectedCumYield = numpy.array([])
+        numpy.testing.assert_equal(excpectedIndYield, actualIndYield)
+        numpy.testing.assert_equal(excpectedCumYield, actualCumYield)
+
+    def test_getXS(self):
+        """Test the getXS method."""
+        # raises exception due to non-existing universes
+        with self.assertRaises(SerpentToolsException):
+            vals, unc = self.reader.getXS('2', 982490, 18, 0)
+        # raises exception due to wrong isotope formatting
+        with self.assertRaises(SerpentToolsException):
+            vals, unc = self.reader.getXS('0', 5000000, 18, 0)
+
+        # Test that values match
+        univ = '0'
+        expectedXSdata = {(10010, 102, 0), (982490, 18, 0), (982510, 102, 0),
+                            (982510, 16, 0), (982510, 17, 0), (982510, 18, 0),
+                            (40090, 107, 0)}
+        expectedValue = [3.09753E-05, 3.33901E-05,
+                           3.57054E-05, 3.70926E-05, 3.61049E-05, 3.39464E-05,
+                           3.39767E-05, 3.98315E-05, 5.38962E-05, 7.96923E-05,
+                           1.18509E-04, 1.73915E-04, 2.54571E-04, 3.38540E-04,
+                           4.52415E-04, 5.98190E-04, 7.69483E-04, 1.04855E-03,
+                           1.31149E-03, 1.67790E-03, 2.15195E-03, 2.70125E-03,
+                           3.44635E-03, 5.04611E-03]
+        expectedUnc = [0.00011, 0.00002, 0.00001, 0.00000, 0.00000, 0.00000,
+                       0.00000, 0.00001, 0.00001, 0.00002, 0.00002, 0.00002,
+                       0.00002, 0.00001, 0.00001, 0.00002, 0.00002, 0.00003,
+                       0.00002, 0.00003, 0.00004, 0.00005, 0.00017, 0.00069]
+        expectedKeys = set(expectedXSdata)
```

Comment:
`expectedXSData` is already a set, similar to above comment.

Reply:
converted to numpy array.